research/2022-07-15-relay-anonymity.md

```diff
@@ -0,0 +1,324 @@
+Because the attacker will receive same messages from both $v$ and its peers, it can leverage timing information.
+Messages that are received significantly faster from $v$ are more likely messages that $v$ sent.
+The attacker can (periodically) measure latency between itself and $v$, and between itself and the peers of $v$ to get more accurate estimates for the expected timings.
+An AS attacker (and if the topology allows, even a local attacker) could also learn the latency between $v$ and its well-behaving peers.
+
+### Controlled Neighbourhood
+
+If a multi node attacker manages to control all peers of the victim node, it can trivially tell which messages originated from $v$.
+
+### Observing Messages
+
+If Waku relay was not protected with Noise, the AS attacker could simply check for messages leaving $v$ which have not been relayed to $v$.
+These are the messages sent by $v$.
+Waku relay protects against this attack by employing Noise.
+
+### Correlation
+
+Monitoring all traffic (in an AS or globally), allows the attacker to identify traffic correlated with messages originating from $v$.
+This (alone) does not allow an external attacker to learn which message $v$ sent, but it allows identifying the respective traffic propagating through the network.
+The more traffic in the network, the lower the success rate of this attack.
+
+Combined with just a few nodes controlled by the attacker, the actual message associated with the correlated traffic can eventually be identified.
+
+### DoS
+
+An active single node attacker could run a disruption attack by
+
+* (1) dropping message that should be relayed
+* (2) flooding neighbours with bogus messages
+
+While (1) has a negative effect on availability, the impact is not significant.
+A linearly scaling botnet attacker, however, could significantly disrupt the network with such an attack.
+(2) is thwarted by [RLN](https://rfc.vac.dev/spec/17/).
+Also [SWAP](https://rfc.vac.dev/spec/18/) helps mitigating DoS attacks.
+
+A local attacker can DoS Waku by dropping all Waku traffic within its controlled network segment.
+An AS attacker can DoS Waku within its authority, while a global attacker can DoS the whole network.
+A countermeasure are censorship resistance techniques like [Pluggable Transports](https://www.pluggabletransports.info/about/).
+
+## Summary and Future Work
+
+Currently, Waku v2 relay offers k-anonymity with respect to receiver anonymity.
+This also includes k-anonymity towards legitimate members of the same topic.
+
+Waku v2 relay offers sender anonymity in the single node attacker model with its [strict no sign policy](https://rfc.vac.dev/spec/11/#signature-policy).
+Currently, Waku v2 does not guarantee sender anonymity in the multi node and stronger attacker models.
+
+We plan to work mix-net integration, which will allow Waku to be anonymity-preserving even within the strongest attacker models.
```

Comment:
I'd flip this paragraph around, and phrase it more in terms of modular privacy building blocks depending on requirements. Dandellion might make sense to mention before mixnets too, instead of leading with mixnet work.

Reply:
in dc0c7c77b